src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -445,6 +452,18 @@ public void delete(long exerciseId, boolean deleteStudentReposBuildPlans, boolea
         }
         else {
             exerciseRepository.delete(exercise);
+            // delete text assessment knowledge if exercise is of type TextExercise and if no other exercise uses same knowledge
+            if (exercise instanceof TextExercise) {
+                log.info("Deleting knowledge of exercise");
+
+                textAssessmentKnowledgeService.deleteKnowledge(((TextExercise) exercise).getKnowledge().getId());
+            }
+            // delete model assessment knowledge if exercise is of type ModelExercise and if no other exercise uses same knowledge
+            if (exercise instanceof ModelingExercise) {
```

Comment:
I think "else if" would be better here since exercise cannot be an instance of both exercise types. If you use else if will save the second if check if the exercise is instance of text exercise

Reply:
Thank you for your feedback @MerlinMehmed 🙂 I incorporated your change!